File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/options/SimpleNumberOption.java
package dev.ragnarok.fenrir.fragment.search.options;

import android.os.Parcel;

import androidx.annotation.NonNull;

import java.util.Objects;

import dev.ragnarok.fenrir.util.ParcelUtils;

public class SimpleNumberOption extends BaseOption {

    public static final Creator<SimpleNumberOption> CREATOR = new Creator<SimpleNumberOption>() {
        @Override
        public SimpleNumberOption createFromParcel(Parcel in) {
            return new SimpleNumberOption(in);
        }

        @Override
        public SimpleNumberOption[] newArray(int size) {
            return new SimpleNumberOption[size];
        }
    };
    public Integer value;

    public SimpleNumberOption(int key, int title, boolean active) {
        super(SIMPLE_NUMBER, key, title, active);
    }

    public SimpleNumberOption(int key, int title, boolean active, int value) {
        super(SIMPLE_NUMBER, key, title, active);
        this.value = value;
    }

    protected SimpleNumberOption(Parcel in) {
        super(in);
        value = ParcelUtils.readObjectInteger(in);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        ParcelUtils.writeObjectInteger(dest, value);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;

        SimpleNumberOption that = (SimpleNumberOption) o;
        return Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + (value != null ? value.hashCode() : 0);
        return result;
    }

    @NonNull
    @Override
    public SimpleNumberOption clone() throws CloneNotSupportedException {
        SimpleNumberOption clone = (SimpleNumberOption) super.clone();
        clone.value = value;
        return clone;
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/fragment/search/options/SimpleNumberOption.kt
package dev.ragnarok.fenrir.fragment.search.options

import android.os.Parcel
import android.os.Parcelable
import dev.ragnarok.fenrir.util.ParcelUtils.readObjectInteger
import dev.ragnarok.fenrir.util.ParcelUtils.writeObjectInteger

class SimpleNumberOption : BaseOption {
    @JvmField
    var value: Int? = null

    constructor(key: Int, title: Int, active: Boolean) : super(SIMPLE_NUMBER, key, title, active)
    constructor(key: Int, title: Int, active: Boolean, value: Int) : super(
        SIMPLE_NUMBER,
        key,
        title,
        active
    ) {
        this.value = value
    }

    internal constructor(`in`: Parcel) : super(`in`) {
        value = readObjectInteger(`in`)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        writeObjectInteger(dest, value)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val that = other as SimpleNumberOption
        return value == that.value
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + if (value != null) value.hashCode() else 0
        return result
    }

    @Throws(CloneNotSupportedException::class)
    override fun clone(): SimpleNumberOption {
        val clone = super.clone() as SimpleNumberOption
        clone.value = value
        return clone
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SimpleNumberOption> {
        override fun createFromParcel(parcel: Parcel): SimpleNumberOption {
            return SimpleNumberOption(parcel)
        }

        override fun newArray(size: Int): Array<SimpleNumberOption?> {
            return arrayOfNulls(size)
        }
    }
}